app/src/main/java/org/dhis2/usescases/main/program/ProgramPresenter.kt

```diff
@@ -99,110 +79,42 @@ class ProgramPresenter internal constructor(
         )
     }
 
-    override fun dispose() {
-        compositeDisposable!!.clear()
+    fun onSyncStatusClick(program: ProgramViewModel) {
+        view.showSyncDialog(program)
     }
 
-    override fun onSyncStatusClick(program: ProgramViewModel) {
-        view!!.showSyncDialog(
-            SyncStatusDialog.Builder()
-                .setConflictType(
-                    if (program.typeName() != "DataSets") {
-                        SyncStatusDialog.ConflictType.PROGRAM
-                    } else {
-                        SyncStatusDialog.ConflictType.DATA_SET
-                    }
-                )
-                .setUid(program.id())
-                .onDismissListener(
-                    object : GranularSyncContracts.OnDismissListener {
-                        override fun onDismiss(hasChanged: Boolean) {
-                            if (hasChanged) {
-                                programQueries.onNext(
-                                    Pair.create(
-                                        FilterManager.getInstance().periodFilters,
-                                        FilterManager.getInstance().orgUnitUidsFilters
-                                    )
-                                )
-                            }
-                        }
-                    })
-                .build()
+    fun updateProgramQueries() {
+        programQueries.onNext(
+            Pair.create(filterManager.periodFilters, filterManager.orgUnitUidsFilters)
         )
     }
 
-    override fun onItemClick(programModel: ProgramViewModel) {
-        val bundle = Bundle()
-        val idTag = if (programModel.typeName() == "DataSets") {
-            "DATASET_UID"
-        } else {
-            "PROGRAM_UID"
-        }
-
-        if (!isEmpty(programModel.type())) {
-            bundle.putString("TRACKED_ENTITY_UID", programModel.type())
-        }
-
-        view!!.analyticsHelper().setEvent(
-            TYPE_PROGRAM_SELECTED,
-            if (programModel.programType().isNotEmpty()) {
-                programModel.programType()
-            } else {
-                programModel.typeName()
-            },
-            SELECT_PROGRAM
-        )
-        bundle.putString(idTag, programModel.id())
-        bundle.putString(Constants.DATA_SET_NAME, programModel.title())
-        bundle.putString(
-            Constants.ACCESS_DATA,
-            java.lang.Boolean.toString(programModel.accessDataWrite())
-        )
-        val programTheme = ColorUtils.getThemeFromColor(programModel.color())
-        val prefs = view!!.abstracContext.getSharedPreferences(
-            Constants.SHARE_PREFS, Context.MODE_PRIVATE
-        )
```

Comment:
very nice dependency decoupling. For next time, I would have injected the analytics controller and track in the presenter